ide/che-core-ide-app/src/main/java/org/eclipse/che/ide/actions/ExportConfigAction.java

```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2015 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.ide.actions;
+
+import com.google.inject.Inject;
+
+import org.eclipse.che.api.analytics.client.logger.AnalyticsEventLogger;
+import org.eclipse.che.ide.CoreLocalizationConstant;
+import org.eclipse.che.ide.api.action.AbstractPerspectiveAction;
+import org.eclipse.che.ide.api.action.ActionEvent;
+import org.eclipse.che.ide.api.app.AppContext;
+import org.eclipse.che.ide.api.app.CurrentProject;
+
+import javax.inject.Singleton;
+import javax.validation.constraints.NotNull;
+import java.util.Arrays;
+
+import static org.eclipse.che.ide.workspace.perspectives.project.ProjectPerspective.PROJECT_PERSPECTIVE_ID;
+
+/**
+ * Exports workspace config.
+ *
+ * @author Sergii Leschenko
+ * @author Max Shaposhnik
+ */
+@Singleton
+public class ExportConfigAction extends AbstractPerspectiveAction {
+    private final AppContext                  appContext;
+    private final AnalyticsEventLogger        eventLogger;
+    private final String                      exportConfigURL;
+
+    @Inject
+    public ExportConfigAction(CoreLocalizationConstant locale,
+                              AppContext appContext,
+                              AnalyticsEventLogger eventLogger) {
+        super(Arrays.asList(PROJECT_PERSPECTIVE_ID), locale.exportConfigText(), null, null, null);
+        this.appContext = appContext;
+        this.eventLogger = eventLogger;
+        this.exportConfigURL = "/api/workspace/" + appContext.getWorkspaceId();
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        eventLogger.log(this);
+        downloadFile(exportConfigURL, appContext.getWorkspace().getName());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void updateInPerspective(@NotNull ActionEvent event) {
+        CurrentProject activeProject = appContext.getCurrentProject();
+        event.getPresentation().setEnabled(activeProject != null);
+    }
+
+    /*
+    * Trick to force browser download the file instead of opening json on new tab
+    */
+    private native void downloadFile(String url, String wsName)/*-{
```

Comment:
on 3.x factory export
we had header for that
https://github.com/codenvy/che-core/blob/master/platform-api/che-core-api-factory/src/main/java/org/eclipse/che/api/factory/server/FactoryService.java#L573-L575

$ curl -i https://codenvy.com/api/factory/workspacewdlne95weroleom2/MovieFun-TomEE
HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Set-Cookie: JSESSIONID=6092FDFBC6539F5D5618A53FF686D1FC; Path=/api/; Secure; HttpOnly
ETag: "d78afc0ba5324533b75d40c325928bf4"
Content-Disposition: attachment; filename=MovieFun-TomEE.json
Cache-Control: public, no-cache, no-store, no-transform
Content-Type: application/json
Content-Length: 929
Date: Wed, 06 Jan 2016 10:22:19 GMT